src/core_modules/capture-core/components/WidgetEventEdit/EditEventDataEntry/editEventDataEntry.epics.js

```diff
@@ -190,3 +193,23 @@ export const requestDeleteEventDataEntryEpic = (action$: InputObservable, store:
             return startDeleteEventDataEntry(eventId, params);
         }));
 
+export const startCreateNewAfterCompletingEpic = (
+    action$: InputObservable, store: ReduxStore, { querySingleResource, history }: ApiUtils) =>
+    action$.pipe(
+        ofType(
+            actionTypes.START_CREATE_NEW_AFTER_COMPLETING,
+            newEventWidgetActionTypes.START_CREATE_NEW_AFTER_COMPLETING,
+        ),
+        flatMap((action) => {
+            const { isCreateNew, enrollmentId } = action.payload;
+            return from(querySingleResource({ resource: 'tracker/enrollments', id: enrollmentId })).pipe(
+                switchMap(({ trackedEntity, program, orgUnit: orgUnitId }) => {
+                    const params = { enrollmentId, orgUnitId, programId: program, teiId: trackedEntity };
```

Comment:
Hey,
According to the Jira description, if there is only one possible stage available, the user should be saved a click and taken directly to the New event workspace for that stage. For this use case, the `stageId` can be added to the URL parameters to achieve the functionality. WDYT?
Thank you!